bugfixtemplates/max-or-min-using-stream-sort/FindMaxOrMinUsingStreamSort.java

```diff
@@ -0,0 +1,29 @@
+package bugfixtemplates;
+
+import java.util.Collection;
+import java.util.stream.Stream;
+
+class CheckAnyMatchUsingFindFirst {
+    
+    Collection<String> col;
+
+    void positiveStreamSortThenOnlyFindFirst() {
+        col.stream().sorted(null).findFirst();
+    }
+
+    void negativeSortThenDoNotFindFirst() {
+        Stream<String> sorted = col.stream().sorted(null);
+        sorted.collect(null);
+    }
+
+    void positiveSortExistingStreamThenFindFirst(Stream<String> stream) {
+        stream.sorted(null).findFirst();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It is more efficient to directly use Stream::min or Stream::max rather than Stream::sorted and Stream::findFirst.  The former has O(n) time complexity while the latter has higher time complexity.